app/Repositories/Summit/DoctrineSpeakerRepository.php

```diff
@@ -30,6 +30,53 @@ final class DoctrineSpeakerRepository
     extends SilverStripeDoctrineRepository
     implements ISpeakerRepository
 {
+    private function buildHasPresentationSubQuery(Filter $filter)
+    {
+        $list_in_conditions = [];
+
+        if ($filter->hasFilter("has_accepted_presentations") &&
+            $filter->getFilter("has_accepted_presentations")[0]->getValue() == "true") {
+            $list_in_conditions[] = "'accepted'";
+        }
+
+        if ($filter->hasFilter("has_alternate_presentations") &&
+            $filter->getFilter("has_alternate_presentations")[0]->getValue() == "true") {
+            $list_in_conditions[] = "'alternate'";
+        }
+
+        if ($filter->hasFilter("has_rejected_presentations") &&
+            $filter->getFilter("has_rejected_presentations")[0]->getValue() == "true") {
+            $list_in_conditions[] = "'unaccepted'";
+        }
+
+        if (count($list_in_conditions) == 0) return "";
+
+        $in_condition = join(",", $list_in_conditions);
+
+        return " AND EXISTS (
+                    SELECT *
+                    FROM (
+                          SELECT CASE
+                                     WHEN CustomOrder IS NULL THEN 'unaccepted'
+                                     WHEN CustomOrder <= SessionCount THEN 'accepted'
+                                     ELSE 'alternate'
+                                 END AS SelectionStatus
+                          FROM (
+                                   SELECT p2.CustomOrder AS CustomOrder, pc.SessionCount AS SessionCount
+                                   FROM Presentation p2
+                                            LEFT JOIN SummitSelectedPresentation sp ON sp.PresentationID = p2.ID
+                                            LEFT JOIN SummitSelectedPresentationList l ON sp.SummitSelectedPresentationListID = l.ID
+                                            LEFT JOIN PresentationCategory pc on l.CategoryID = pc.ID
+                                   WHERE p2.ID = P.ID AND sp.Collection = 'selected'
+                                   UNION
+                                   SELECT null, null
```

Comment:
why is this ?

Reply:
It's a fallback. If no rows are returned from the first part of the union, the second part will act as a dummy to fulfill the CASE WHEN -> 'unaccepted' condition

Reply:
subquery adjusted and code comments added